crates/agent/src/agent_diff.rs

```diff
@@ -380,7 +380,10 @@ fn update_editor_selection(
     window: &mut Window,
     cx: &mut Context<Editor>,
 ) {
-    let newest_cursor = editor.selections.newest::<Point>(cx).head();
+    let newest_cursor = editor
+        .selections
+        .newest::<Point>(&editor.selections.display_map(cx))
```

Comment:
@RemcoSmitsDev nitpick, but lets add `editor.display_snapshot(cx)` to avoid the need for all these callers to reach into the selections object twice.